plugins/UsersManager/Controller.php

```diff
@@ -387,37 +387,67 @@ public function addNewToken()
             throw new Exception('Not allowed');
         }
 
-        $noDescription = false;
+        $invalidExpireDate = null;
+        if (!empty($_POST['token_expire_date'])) {
+            $tokenExpireDate = \Piwik\Request::fromRequest()->getStringParameter('token_expire_date');
+
+            try {
+                $invalidExpireDate = true;
+                if (preg_match('/^(\d{4})-(\d{2})-(\d{2})$/', $tokenExpireDate)) {
+                    Date::factory($tokenExpireDate);
+                    $invalidExpireDate = false;
+                }
+            } catch (Exception $e) {
+                // nop
+            }
+        }
 
+        $noDescription = null;
+        $description = '';
         if (!empty($_POST['description'])) {
```

Comment:
As mentioned above: we should never directly access `$_POST`. Better to use `Request::fromPost`.

Reply:
Updated to use Request::fromPost via a variable